### DISCUSSION TITLE
Is there a way to change the standard component separator?

### DISCUSSION THREAD

**Discussion by GordonMegaDeath** [nextgenhealthcare/connect#6320] [Q&A]
Asking out of curiosity. 

Standard separator is ^ IE FRESH^DOUG^E

Is there a way to change the ^ to ~ across the entire message?

I've tried message.replace(/\^/g, '~');  in the preprocessor scripts but that has no change. 

any help would be greatly appreciated

**Comment by pacmano1** [comment]
```JS
return message.replace(/\^/g,'~')
```

Sorta gotta wonder why?

![image](https://github.com/user-attachments/assets/afcca16c-bc6a-41d5-b5f8-c38383be487c)

![image](https://github.com/user-attachments/assets/d0fa03c7-ce8c-4ba1-965f-6435200d7691)

**Reply by GordonMegaDeath** [reply]
Had a site claim a 3rd party vendor has problems with ^ separator(seems fishy) and they asked if we could replace it with ~.  Pretty sure the answer is going to wind up being 'NO', but I got curious as to how it could be done.

I appreciate your reply.

**Reply by jonbartels** [reply]
> Had a site claim a 3rd party vendor has problems with ^ separator(seems fishy)

Yes it is fishy, but the HL7 standard is designed to support this. I remember that if you receive HL7 with non-standard separators that Mirth will automatically parse it. Are you receiving HL7, sending HL7, or both?

@pacmano1 suggests an OK solution but I think there is a better way to support this in Mirth. It's HL7 renderer should inherently support parsing MSH.1 correctly. 

I searched for documentation:
https://docs.nextgen.com/bundle/Mirth_User_Guide_4_5_0/page/connect/connect/topics/c_HL7_v2_x_Data_Type_connect_ug.html

I searched the old forums, using Google because the Forum search is horrible `site:forums.mirthproject.io yoursearch terms` 
https://www.google.com/search?q=site%3Aforums.mirthproject.io+default+separator+hl7
https://google.com/search?q=site%3Aforums.mirthproject.io+separator+hl7

*I think this might be a solution* https://forums.mirthproject.io/forum/mirth-connect/support/13329-having-issues-with-a-non-standard-hl7-delimiter -> See how Nick just changes the MSH.2 value? Internally Mirth is generating HL7 as XML. If MSH.2 is set with non-standard delimiters then when Mirth renders that XML to HL7 (LLP technically) it will use the separators defined in MSH.2

**Reply by pacmano1** [reply]
I was just fixing the replace, lol.  But I'd be on the call with the other side wondering if they really know what they are doing for a living.
